Release Notes Writing Guide (#8969)
desktop/desktop: docs/process/writing-release-notes.md

## Anatomy of a Release Note 👩🏼‍⚕️

A release note has three (or four) parts.

```
[Tag] Description of work or change - #{issue_number}
```

If it was done by an external contributor, we'll add an extra part — a thank you!

```
[Tag] Description of work or change - #{issue_number}. Thanks @{contributor_username}!
```

## Writing Style 🖊️ 

These mostly apply to writing the release note description itself.

### User Facing

We don't include release notes that don't impact the user's experience using Desktop. For example, we wouldn't include a release not about fixing our Appveyor CI config or upgrading Electron. We'll make exceptions for security vulnerability fixes, especially if they're high profile.

### User Impact

We describe our work in terms of impact on our users, not technical process. How does this work change the user's workflow or experience? What does it help them do?

For example, we might say:

```
[Fixed] Keep PR badge on top of progress bar - #8622
```

but **not**:

```
[Fixed] Increase z-index of the progress bar PR badge - #8622
```

### Present Tense

> We can never remember if we're supposed to active, present, or past tense, but I think we're mostly using present.

Reviewer: ```suggestion
We attempt to use present tense unless it significantly reduces clarity.
```

For example:

```
[Fixed] Fix arrow key navigation of Changes and History lists - #6845
```

but **not** this

```
[Added] Adding integration for Xcode as external editor - #8255
```

## Tags 🛄

> We use four different tags to organize our release notes: [New], [Added], [Fixed], [Improved], [Removed]. Here's when to use which one.

Author: ```suggestion
We use four different tags to organize our release notes: `[New]`, `[Added]`, `[Fixed]`, `[Improved]`, `[Removed]`. Here's when to use which one.
```

> We use four different tags to organize our release notes: `[New]`, `[Added]`, `[Fixed]`, `[Improved]`, `[Removed]`. Here's when to use which one.

Reviewer 2: ```suggestion
We use five different tags to organize our release notes: `[New]`, `[Added]`, `[Fixed]`, `[Improved]`, `[Removed]`. Here's when to use which one.
```

Author: hahaha 🙃
![math](https://media.giphy.com/media/XyOrJljDNBEpa/giphy.gif)

These aren't hard and fast rules or categories, 🙃 so utilize with your own judgment and nuance.

### [New]

> [New] is usually reserved for our shiniest features. The note itself might be short and very high level and encapsulate a _lot_ of development work.

Reviewer 3: ```suggestion
[New] is usually reserved for our shiniest features. The note itself might be short and very high level and encapsulate a _lot_ of development work.

A great way to check if we have all the [New] lines we need is to ensure that the lines match the "highlights" or "most important new features" of that given release.
```

### [Added]

[Added] is basically [New] but for a smaller feature. Or its just not something that's highlighted as much as the [New] things in this release.

> We often use [Added] for new editor and terminal integrations.

Reviewer 2: This example is super helpful 👍

Author: oo that's good feedback, i'll add more examples to the rest of these!

### [Fixed]

Good ol' [Fixed], the bread and butter of release notes. We use this to indicate that something was broken but now it's not!

### [Improved]

Somewhere in between [Added] and [Fixed] lies [Improved]. The idea is that we've made an existing feature or part of the app better, but it wasn't necessarily broken before. Think of it like an "enhancement" release note.

> Does this sound like [Added]? Is that confusing? Yes.

Reviewer 2: 🤣

> Does this sound like [Added]? Is that confusing? Yes.

Reviewer 3: ```suggestion
Does this sound like [Added]? Is that confusing? Yes. The rule of thumb is that if it's a new end-to-end piece of functionality that's small, it's [Added]. If it's a change to a portion of a feature, it's [Improved]
```

### [Removed]

> We rarely use this. Historically, I think it's been primarily used when we've unshipped a feature.

Reviewer: ```suggestion
This describes a feature that is no longer available in the app. We rarely use it.
```

> 

Reviewer 3: We should probably put something here about putting the release notes in the right order. That's stuff that exists in the releasing-updates.md doc but it might make sense to extract and put here instead.

Author: also done in 7e4e08c
